fix: format table name to include backticks

jorgerojas26 opened the pull request:
Fixes #47

ccoVeille commented on drivers/mysql.go:
```diff
@@ -313,6 +320,7 @@ func (db *MySQL) ExecuteQuery(query string) (results [][]string, err error) {
 // TODO: Rewrites this logic to use the primary key instead of the id
 func (db *MySQL) UpdateRecord(table, column, value, primaryKeyColumnName, primaryKeyValue string) error {
+	table = db.formatTableName(table)
 	query := fmt.Sprintf("UPDATE %s SET %s = \"%s\" WHERE %s = \"%s\"", table, column, value, primaryKeyColumnName, primaryKeyValue)
```
```suggestion
 	query := fmt.Sprintf(`UPDATE %s SET %s = "%s" WHERE %s = "%s"`, table, column, value, primaryKeyColumnName, primaryKeyValue)
```

ccoVeille replied:
Also, I'm very worried about the  possible SQL injection

You should consider using `?` for value, then pass `value ` to Exec as an arg

ccoVeille commented on drivers/mysql.go:
```diff
@@ -321,6 +329,7 @@ func (db *MySQL) UpdateRecord(table, column, value, primaryKeyColumnName, primar
 // TODO: Rewrites this logic to use the primary key instead of the id
 func (db *MySQL) DeleteRecord(table, primaryKeyColumnName, primaryKeyValue string) error {
+	table = db.formatTableName(table)
 	query := fmt.Sprintf("DELETE FROM %s WHERE %s = \"%s\"", table, primaryKeyColumnName, primaryKeyValue)
```
Same concerns here

ccoVeille commented on drivers/mysql.go:
```diff
@@ -385,7 +394,7 @@ func (db *MySQL) ExecutePendingChanges(changes []models.DbDmlChange, inserts []m
 		statementType = "DELETE FROM"
 
-		query = fmt.Sprintf("%s %s WHERE %s = \"%s\"", statementType, delete.Table, delete.PrimaryKeyColumnName, delete.PrimaryKeyValue)
+		query = fmt.Sprintf("%s %s WHERE %s = \"%s\"", statementType, db.formatTableName(delete.Table), delete.PrimaryKeyColumnName, delete.PrimaryKeyValue)
```
Same

ccoVeille commented on drivers/mysql.go:
```diff
@@ -405,7 +414,7 @@ func (db *MySQL) ExecutePendingChanges(changes []models.DbDmlChange, inserts []m
 		}
 
-		query := fmt.Sprintf("INSERT INTO %s (%s) VALUES (%s)", insert.Table, strings.Join(insert.Columns, ", "), strings.Join(values, ", "))
+		query := fmt.Sprintf("INSERT INTO %s (%s) VALUES (%s)", db.formatTableName(insert.Table), strings.Join(insert.Columns, ", "), strings.Join(values, ", "))
```
Here too the values should be passed as ?,?,?...

strings.Repeat could help

And `values` should be passed to Exec

ccoVeille commented on drivers/mysql.go:
```diff
@@ -442,3 +449,18 @@ func (db *MySQL) SetProvider(provider string) {
+	splittedTableName := strings.Split(tableName, ".")
+
+	if len(splittedTableName) == 1 {
+		return tableName
```
You could return this


```suggestion
		return fmt.Sprintf("`%s`", table)
```

ccoVeille reviewed (commented):
I'm sorry because most of my feedbacks are  not related to the current PR.

But I think I had to report them

jorgerojas26 commented:
Thanks for reviewing the PR. May i ask why SQL Injection is a concern to you in this app when it is just a SQL client?.

I totally understand that i should use prepared statements, i plan to make a huge refactor on all of the queries, but i want to solve the problem with a quick solution first. Does it makes sense to you?

jorgerojas26 commented:
The table name formatting is really dirty, but it is a quick solution. That should be refactored too.

ccoVeille commented:
> Thanks for reviewing the PR. May i ask why SQL Injection is a concern to you in this app when it is just a SQL client?.

I tend to always plan the worst. Often, I'm not disappointed and something close happens 😅

> I totally understand that i should use prepared statements, i plan to make a huge refactor on all of the queries, but i want to solve the problem with a quick solution first. Does it makes sense to you?

Not necessarily prepared statement, but using Exec the right way should be always preferred.

ccoVeille commented:
Also please consider a table could have a varchar primary key, and that this field may contain a quote

Something like `john "babayaga" wick`

It would be a bad design to rely on a varchar PK for this, but your app will be unable to work with this row

ccoVeille commented:
> May i ask why SQL Injection is a concern to you in this app when it is just a SQL client?

My SQL injection I was more thinking about a source of bugs, than a attack.

Maybe it's what you were asking for

jorgerojas26 commented:
> > May i ask why SQL Injection is a concern to you in this app when it is just a SQL client?
> 
> My SQL injection I was more thinking about a source of bugs, than a attack.
> 
> Maybe it's what you were asking for

I understand. I appreciate the feedback. For now, i'll merge this PR just to resolve the connection problem and open another PR addressing your feedback.

Thank you very much.

ccoVeille commented:
Sure
